src/hotspot/share/gc/z/zStat.cpp

```diff
@@ -808,7 +808,7 @@ ZStatSubPhase::ZStatSubPhase(const char* name, ZGenerationId id)
   : ZStatPhase(id == ZGenerationId::young ? "Young Subphase" : "Old Subphase", name) {}
 
 void ZStatSubPhase::register_start(ConcurrentGCTimer* timer, const Ticks& start) const {
-  if (timer != nullptr) {
+  if (timer != nullptr && !ZAbort::should_abort()) {
```

Comment:
`register_end` also skips the logging part if we should abort. Should we do the same for the `register_start`?

Reply:
My initial implementation had that. (just an early return)
But thought it might be interesting to see if there ever is an issue with the abort code to have the phase starts in the log. The end logging is less interesting as we are aborting. This is how single generation also does it.
Given that we only log sub phases on debug, maybe this is less useful as we will miss it in the normal gc* logs.